pkg/kubeadm/master/pki.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubemaster
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"net"
+	"path"
+
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+	"k8s.io/kubernetes/pkg/kubeadm/tlsutil"
+	"k8s.io/kubernetes/pkg/util/crypto"
+)
+
+func newCertificateAuthority() (*rsa.PrivateKey, *x509.Certificate, error) {
+	key, err := tlsutil.NewPrivateKey()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	config := tlsutil.CertConfig{
+		CommonName: "kubernetes",
+	}
+
+	cert, err := tlsutil.NewSelfSignedCACertificate(config, key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return key, cert, err
+}
+
+func newServerKeyAndCert(caCert *x509.Certificate, caKey *rsa.PrivateKey, altNames tlsutil.AltNames) (*rsa.PrivateKey, *x509.Certificate, error) {
+	key, err := tlsutil.NewPrivateKey()
+	if err != nil {
+		return nil, nil, err
+	}
+	altNames.IPs = append(altNames.IPs, net.ParseIP("10.3.0.1"))
+	altNames.DNSNames = append(altNames.DNSNames,
+		"kubernetes",
+		"kubernetes.default",
+		"kubernetes.default.svc",
+		"kubernetes.default.svc.cluster.local",
```

Comment:
Yes, but I don't think we need to provide a flag that will show up in `--help`, so I'd go with environment variables for now. However, whatever we will do, how should user express those? May be we could just append extra SANs and keep these as they are?